JAVA1/src/Solution/FileSearchInHomeDirectory.java

```diff
@@ -0,0 +1,54 @@
+package Solution;
+import java.io.File;
+import java.util.*;
+import java.util.regex.*;
+public class FileSearchInHomeDirectory {
+
+    private List<File> mResultArray =new ArrayList<>();
+
+    /**
+     * getter function to get the mResultArray that is list of all matching files
+     * @return Returns list of all matching files.
+     */
+    public List<File> getmResultArray(){
+        return mResultArray;
+    }
+    /**
+     * Prints the complete path if file matching regex is found. It searches in all directories and
+     * subdirectories recursively.
+     * @param rgx The regular expression that has to be matched
+     * @param directory The home directory where search is carried out
+     */
+    public boolean findFiles(String rgx, String directory){
+        try{
+            Pattern.compile(rgx);
+        }catch (PatternSyntaxException |NullPointerException e){
```

Comment:
NullPointerException must not have exception handling. Null check must be added wherever necessary.